src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -3531,6 +3519,177 @@ private bool NotesAlreadyLoaded(IEnumerable<ExtraAnnotationViewInfo> notes)
         }
         #endregion
 
+        #region Insert Private Methods
+
+        /// <summary>
+        /// Update inserted workspace elements GUIDs 
+        /// </summary>
+        /// <param name="viewInfo">viewInfo</param>
+        /// <param name="ws">workspace</param>
+        private void UpdateWorkspaceGUIDs(ref ExtraWorkspaceViewInfo viewInfo, ref WorkspaceModel ws)
+        {
+            var nodes = ws.Nodes;
+            var connectors = ws.Connectors;
+            var notes = viewInfo?.Annotations;
+
+            var nodesGuids = UpdateNodeGUIDs(ref nodes);
+            var connectorsGuids = UpdateConnectorGUIDs(ref connectors);
+            var notesGuids = UpdateNoteGUIDs(nodesGuids, ref notes);
+
+            // Now propagate these changes to the viewInfo object
+            if (viewInfo == null) return;
+
+            PropagateUpdatesToViewInfo(viewInfo, nodesGuids, notesGuids, connectorsGuids);
+        }
+
+        private Dictionary<string, string> UpdateNoteGUIDs(Dictionary<Guid, Guid> nodesGuids, ref IEnumerable<ExtraAnnotationViewInfo> notes)
```

Comment:
again, I see no need for `ref` - please explain.

Reply:
Yup, got it!